test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -4798,6 +4798,70 @@ BOOST_AUTO_TEST_CASE(byte_array_push)
 	ABI_CHECK(callContractFunction("test()"), encodeArgs(false));
 }
 
+BOOST_AUTO_TEST_CASE(array_pop)
+{
+	char const* sourceCode = R"(
+		contract c {
+			uint[] data;
+			function test() public returns (uint x, uint y) {
+				data.push(7);
+				x = data.push(3);
+				data.pop();
+				y = data.push(2);
+			}
+		}
+	)";
+	compileAndRun(sourceCode);
+	ABI_CHECK(callContractFunction("test()"), encodeArgs(2, 2));
+}
+
+BOOST_AUTO_TEST_CASE(array_pop_empty)
+{
+	char const* sourceCode = R"(
+		contract c {
+			uint[] data;
+			function test() public returns (bool) {
+				data.pop();
+				return true;
+			}
+		}
+	)";
+	compileAndRun(sourceCode);
+	ABI_CHECK(callContractFunction("test()"), encodeArgs());
+}
+
+BOOST_AUTO_TEST_CASE(bytearray_pop)
+{
+    char const* sourceCode = R"(
+        contract c {
+            bytes data;
+            function test() public returns (uint x, uint y) {
+                data.push(7);
+                x = data.push(3);
+                data.pop();
+                data.pop();
+                y = data.push(2);
+            }
+        }
+    )";
+    compileAndRun(sourceCode);
+    ABI_CHECK(callContractFunction("test()"), encodeArgs(2, 1));
+}
+
+BOOST_AUTO_TEST_CASE(bytearray_pop_empty)
+{
+	char const* sourceCode = R"(
+		contract c {
+			bytes data;
+			function test() public returns (bool) {
+				data.pop();
+			}
+		}
+	)";
+	compileAndRun(sourceCode);
+	ABI_CHECK(callContractFunction("test()"), encodeArgs());
+}
+
```

Comment:
Please add tests that

 - push and pop across the 32 byte boundary for byte arrays
 - check that storage is properly cleaned (there are other tests here that do that, search for `storageEmpty`), but for that, you have to reduce the length to zero.

Reply:
Done.